rust/src/dataset/scanner.rs

```diff
@@ -378,26 +378,85 @@ impl Scanner {
             let knn_node = self.ann(q, &index)?; // score, _rowid
             let with_vector = self.dataset.schema().project(&[&q.column])?;
             let knn_node_with_vector = self.take(knn_node, &with_vector)?;
-            let knn_node = if q.refine_factor.is_some() {
-                self.flat_knn(knn_node_with_vector, q)?
+            let mut knn_node = if q.refine_factor.is_some() {
+                self.flat_knn(vec![knn_node_with_vector], q)?
             } else {
                 knn_node_with_vector
             }; // vector, score, _rowid
+
+            // Check if we've created new versions since the index
+            let version = index.dataset_version;
+            if version != self.dataset.version().version {
+                // If we've added more rows, then we'll have new fragments
+                let ds = self.dataset.checkout_version(version).await?;
+                let max_fragment_id_idx = ds
+                    .manifest
+                    .max_fragment_id()
+                    .ok_or_else(|| Error::IO("No fragments in index version".to_string()))?;
+                let max_fragment_id_ds = self
```

Comment:
This does not allow to reclaim / recycle fragment IDs? Could you add a comment? 

In case of a recycle Ids, it seems need to re-mapping all fragment ids.